sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonService.java

```diff
@@ -251,73 +272,100 @@ public Predicate createUserFilter(CriteriaBuilder cb, CriteriaQuery cq, From<?,
-			cb,
-			eventParticipantService.createUserFilter(
-				new EventParticipantQueryContext(cb, cq, joins.getEventParticipantJoins()),
-				new EventUserFilterCriteria().includeUserCaseAndEventParticipantFilter(false).forceRegionJurisdiction(true)),
-			eventParticipantService.createDefaultFilter(cb, joins.getEventParticipant()));
-		final Supplier<Predicate> immunizationFilter = fullImmunizationModuleUsed
-			? () -> CriteriaBuilderHelper.and(
-				cb,
-				immunizationService.createUserFilter(new ImmunizationQueryContext(cb, cq, joins.getImmunizationJoins())),
-				immunizationService.createDefaultFilter(cb, joins.getImmunization()))
-			: () -> null;
-		final Supplier<Predicate> travelEntryFilter = () -> CriteriaBuilderHelper.and(
-			cb,
-			travelEntryService.createUserFilter(new TravelEntryQueryContext(cb, cq, joins.getTravelEntryJoins())),
-			travelEntryService.createDefaultFilter(cb, joins.getTravelEntry()));
-
-		// 2. Define the Joins on associations where needed
-		PersonAssociation personAssociation =
+		/*
+		 * Creates a combined filter only if no filtering on a specific association is given.
+		 */
+		final Predicate userFilter;
+		final PersonAssociation personAssociation =
 			Optional.ofNullable(personCriteria).map(e -> e.getPersonAssociation()).orElse(PersonCriteria.DEFAULT_ASSOCIATION);
 		switch (personAssociation) {
 		case ALL:
-			return CriteriaBuilderHelper.or(
-				cb,
-				caseFilter.get(),
-				contactFilter.get(),
-				eventParticipantFilter.get(),
-				fullImmunizationModuleUsed ? immunizationFilter.get() : null,
-				travelEntryFilter.get());
+			// Combine all associations that are permitted
+			userFilter = CriteriaBuilderHelper.or(
+				queryContext.getCriteriaBuilder(),
+				createAssociationPredicate(queryContext, PersonAssociation.CASE),
+				createAssociationPredicate(queryContext, PersonAssociation.CONTACT),
+				createAssociationPredicate(queryContext, PersonAssociation.EVENT_PARTICIPANT),
+				createAssociationPredicate(queryContext, PersonAssociation.IMMUNIZATION),
+				createAssociationPredicate(queryContext, PersonAssociation.TRAVEL_ENTRY));
+			break;
 		case CASE:
-			return caseFilter.get();
 		case CONTACT:
-			return contactFilter.get();
 		case EVENT_PARTICIPANT:
-			return eventParticipantFilter.get();
 		case IMMUNIZATION:
-			if (!fullImmunizationModuleUsed) {
-				throw new UnsupportedOperationException(
-					"Filtering persons by immunizations is not supported when the reduced immunization module is used.");
-			}
-			return immunizationFilter.get();
 		case TRAVEL_ENTRY:
-			return travelEntryFilter.get();
+			userFilter = createAssociationPredicate(queryContext, personAssociation);
+			break;
 		default:
 			throw new IllegalArgumentException(personAssociation.toString());
 		}
+
+		if (userFilter == null) {
+			throw new IllegalArgumentException("No filter compiled for persons by association " + personAssociation.name());
```

Comment:
Instead of throwing an exception return a "false" predicate.
This will likely also solve the external visit problems.